--- lib/cmake/Qt6Core/Qt6CoreMacros.cmake	2022-11-10 16:08:46.845156002 +0100
+++ lib/cmake/Qt6Core/Qt6CoreMacros.cmake	2022-11-10 16:09:12.735517356 +0100
@@ -548,7 +548,6 @@
     cmake_parse_arguments(PARSE_ARGV 1 arg "MANUAL_FINALIZATION" "" "")
 
     _qt_internal_create_executable("${target}" ${arg_UNPARSED_ARGUMENTS})
-    target_link_libraries("${target}" PRIVATE Qt6::Core)
 
     if(arg_MANUAL_FINALIZATION)
         # Caller says they will call qt6_finalize_target() themselves later
